plugins/modules/cm_service.py

```diff
@@ -0,0 +1,352 @@
+# Copyright 2024 Cloudera, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ansible_collections.cloudera.cluster.plugins.module_utils.cm_utils import (
+    ClouderaManagerModule,
+
+)
+
+from cm_client.rest import ApiException
+from cm_client import MgmtRolesResourceApi
+from cm_client import MgmtServiceResourceApi
+from cm_client import MgmtRoleCommandsResourceApi
+from cm_client import HostsResourceApi
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = r"""
+---
+module: cm_service
+short_description: Interact with Cloudera Manager service roles  
+description:
+  - Create or remove one or more Cloudera Manager service roles.
+  - Start, stop or restart one or more Cloudera Manager service roles.
+author:
+  - "Ronald Suplina (@rsuplina)"
+requirements:
+  - cm_client
+"""
+
+EXAMPLES = r"""
+---
+- name: Create and start Cloudera Manager service roles
+  cloudera.cluster.cm_version:
+    host: "10.10.10.10"
+    username: "jane_smith"
+    password: "S&peR4Ec*re"
+    port: "7180"
+    purge: True
```

Comment:
Can you expand the examples to showcase the various functions? For instance, the use of `purge` to reset the entire state vs selected roles. While these two example do cover a lot of the functions, I suggest expanding the examples to be more scoped in what they are demonstrating.